test/core/validate-graph.test.ts

```diff
@@ -0,0 +1,55 @@
+import { validatePackageURL } from '../../src/core/validate-graph';
+
+describe('validatePackageURL', () => {
+  describe('deb package type tests', () => {
```

Comment:
I'm confused by all these tests, I don't think we want to ever include the source name in the purl's package name.

I thought what we wanted was:
```
name: 'foo/bar'
purl: 'pkg:deb/bar@1.2.3?upstream=foo'
```
(not necessarily correct purl syntax)

All your test cases are *invalid* purls; from the [purl-spec](https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst)
> The namespace is the "vendor" name such as "debian" or "ubuntu". It is not case sensitive and must be lowercased.

And for example in your first test case, `foo` would be the "vendor" name.